### ISSUE TITLE
NullReferenceException in RemoveDuplicatesWithLastOneWinsPolicy.Execute

### ISSUE THREAD

**Issue by bartdesmet** [dotnet/buildtools#1238]
Keep seeing this when trying to run `System.Linq.Expressions` unit tests in `corefx` in VS 2015:

```
1>------ Build started: Project: System.Linq.Expressions.Tests, Configuration: Debug Any CPU ------
1>  System.Linq.Expressions.Tests -> C:\Projects\bartdesmet-corefx\corefx\bin\AnyOS.AnyCPU.Debug\System.Linq.Expressions.Tests\System.Linq.Expressions.Tests.dll
1>C:\Projects\bartdesmet-corefx\corefx\Tools\publishtest.targets(97,5): error MSB4018: The "RemoveDuplicatesWithLastOneWinsPolicy" task failed unexpectedly.
1>C:\Projects\bartdesmet-corefx\corefx\Tools\publishtest.targets(97,5): error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object.
1>C:\Projects\bartdesmet-corefx\corefx\Tools\publishtest.targets(97,5): error MSB4018:    at Microsoft.DotNet.Build.Tasks.RemoveDuplicatesWithLastOneWinsPolicy.Execute()
1>C:\Projects\bartdesmet-corefx\corefx\Tools\publishtest.targets(97,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
1>C:\Projects\bartdesmet-corefx\corefx\Tools\publishtest.targets(97,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
```

**Comment by hughbe** [comment]
I can reproduce this running F5 from a test project

**Comment by weshaggard** [comment]
@naamunds can you please have a look.

**Comment by danmosemsft** [comment]
@naamunds

**Comment by naamunds** [comment]
I'm testing a fix that I've made locally for this issue.  The root cause is that the `TestNugetRuntimeId` property wasn't set when `PrereleaseResolveNuGetPackageAssets` was called, causing TestCopyLocal to be null.

**Comment by naamunds** [comment]
Note: This is only an issue for some test projects because for the projects that run `ResolvePkgProjReferences` the `GetDefaultTestRid` target is run because `ResolvePkgProjReferences` depends on it.  The fix in #1250 is to make `CopyTestToTestDirectory` also depend on `GetDefaultTestRid` so that it's run for every test project.
